### PR TITLE
Adds `tech-remove-intel-xstr` sexp

### PR DESCRIPTION
**Pull request by wookieejedi**
In regards to @MjnMixael's request in #2425, this PR adds the sexp `tech-remove-intel-xstr`, which simply toggles off the specified intel entry.

### CODE REVIEW

File: code/parse/sexp.cpp [hunk 1]
```diff
@@ -13989,6 +13991,42 @@ void sexp_tech_add_intel(int node, bool xstr)
 	}
 }
 
+// wookieejedi, base copied from sexp_tech_add_intel()
+void sexp_tech_remove_intel(int node, bool xstr)
```
Comment (Goober5000) [inline code/parse/sexp.cpp:13995]: Considering that the add and remove operations are almost entirely identical, except for the flags and messages, I would ask that the two functions be combined together and a `bool add` parameter be inserted before the `bool xstr` parameter.  (I'm in the process of combining a bunch of other programs as part of my sexp caching overhaul, but that isn't PR-ready yet.)  Take a look at the dock/undock sexps to see how I handled warning messages for the different sexps.

File: code/parse/sexp.cpp [hunk 2]
```diff
@@ -564,6 +564,7 @@ SCP_vector<sexp_oper> Operators = {
 	{ "tech-add-weapons",				OP_TECH_ADD_WEAPON,						1,	INT_MAX,	SEXP_ACTION_OPERATOR,	},
 	{ "tech-add-intel",					OP_TECH_ADD_INTEL,						1,	INT_MAX,	SEXP_ACTION_OPERATOR,	},	// Goober5000
```
Comment (Goober5000) [inline code/parse/sexp.cpp:565]: Since it's just a few lines of code, I suggest a `tech-remove-intel` sexp be added as well.

**Comment by wookieejedi** [comment]
Thanks for the feedback, both points are incorporated.

File: code/parse/sexp.cpp [hunk 3]
```diff
@@ -13980,12 +13984,17 @@ void sexp_tech_add_intel(int node, bool xstr)
 
 		// if we have an xstr, we already translated this node in the preloader, so just look it up
 		i = intel_info_lookup(name);
-		if (i >= 0)
-			Intel_info[i].flags |= IIF_IN_TECH_DATABASE;
+		if (i >= 0) {
+			if (add) {
+				Intel_info[i].flags |= IIF_IN_TECH_DATABASE;
+			} else {
+				Intel_info[i].flags &= IIF_IN_TECH_DATABASE;
```
Comment (Goober5000) [inline code/parse/sexp.cpp:13991]: I believe this should be `&= ~`.

File: code/parse/sexp.cpp [hunk 4]
```diff
@@ -13980,12 +13984,17 @@ void sexp_tech_add_intel(int node, bool xstr)
...
+			}
+		}
 		else if (xstr)
-			Warning(LOCATION, "In tech-add-intel-xstr, entry XSTR(\"%s\", %d) invalid", name, id);
+			Warning(LOCATION, "In tech-%-intel-xstr, entry XSTR(\"%s\", %d) invalid", (add? "add" : "remove"), name, id);
```
Comment (Goober5000) [inline code/parse/sexp.cpp]: That should be `%s` to specify strings.  `%` by itself won't show anything.  Also, as a nitpick, put a space before the `?`.

File: code/parse/sexp.cpp [hunk 5]
```diff
@@ -13980,12 +13984,17 @@ void sexp_tech_add_intel(int node, bool xstr)
...
 		else
-			Warning(LOCATION, "In tech-add-intel, entry \"%s\" invalid", name);
+			Warning(LOCATION, "In tech-%-intel, entry \"%s\" invalid", (add ? "add" : "remove"), name);
```
Comment (Goober5000) [inline code/parse/sexp.cpp]: Same here with the `%`.

**Comment by wookieejedi** [comment]
Thanks, realized I forgot to add the `s` right before you said it :D

File: code/parse/sexp.h [hunk 6]
```diff
@@ -746,6 +746,8 @@ class waypoint_list;
 #define OP_MODIFY_VARIABLE_XSTR				(0X0030 | OP_CATEGORY_CHANGE2 | OP_NONCAMPAIGN_FLAG)	// m!m
 #define OP_RESET_POST_EFFECTS				(0x0031 | OP_CATEGORY_CHANGE2 | OP_NONCAMPAIGN_FLAG)	// Goober5000
 #define OP_ADD_REMOVE_HOTKEY				(0x0032 | OP_CATEGORY_CHANGE2 | OP_NONCAMPAIGN_FLAG)    // wookieejedi
+#define OP_TECH_REMOVE_INTEL_XSTR			(0x0033 | OP_CATEGORY_CHANGE2 | OP_NONCAMPAIGN_FLAG)    // wookieejedi
+#define OP_TECH_REMOVE_INTEL				(0x0034 | OP_CATEGORY_CHANGE  | OP_NONCAMPAIGN_FLAG)    // wookieejedi
```
Comment (Goober5000) [inline code/parse/sexp.h:750]: `CHANGE2` should be used here, because otherwise it's the same ID as another existing sexp.

**Comment by wookieejedi** [comment]
Ah good catch, thanks!

File: fred2/sexp_tree.cpp [hunk 7]
```diff
@@ -834,6 +834,7 @@ void sexp_tree::right_clicked(int mode)
 							case OP_CUTSCENES_SHOW_SUBTITLE:
 							case OP_ORDER:
 							case OP_TECH_ADD_INTEL:
+							case OP_TECH_REMOVE_INTEL:
```
Comment (Goober5000) [inline fred2/sexp_tree.cpp:837]: Look a few lines up from here where the commented-out `case OP_TECH_ADD_INTEL_XSTR` is.  Add `OP_TECH_REMOVE_INTEL_XSTR` there too.

File: fred2/sexp_tree.cpp [hunk 8]
```diff
@@ -881,6 +882,7 @@ void sexp_tree::right_clicked(int mode)
...
```
Comment (Goober5000) [inline fred2/sexp_tree.cpp:885]: Same thing a few lines up from here.

File: qtfred/src/ui/widgets/sexp_tree.cpp [hunk 9]
```diff
@@ -4954,6 +4954,7 @@ std::unique_ptr<QMenu> sexp_tree::buildContextMenu(QTreeWidgetItem* h) {
 					case OP_CUTSCENES_SHOW_SUBTITLE:
 					case OP_ORDER:
 					case OP_TECH_ADD_INTEL:
+					case OP_TECH_REMOVE_INTEL:
```
Comment (Goober5000) [inline qtfred/src/ui/widgets/sexp_tree.cpp:4957]: Same deal for qtFRED.

File: qtfred/src/ui/widgets/sexp_tree.cpp [hunk 10]
```diff
@@ -5018,6 +5019,7 @@ std::unique_ptr<QMenu> sexp_tree::buildContextMenu(QTreeWidgetItem* h) {
...
```
Comment (Goober5000) [inline qtfred/src/ui/widgets/sexp_tree.cpp:5022]: Again for qtFRED.

**Comment by wookieejedi** [comment]
Ah that's good to know, thanks! Just added.

**Pull request merged**